pymeasure/instruments/hp/hp663X2A.py

```diff
@@ -0,0 +1,233 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2024 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import Instrument, SCPIMixin
+from pymeasure.instruments.validators import strict_discrete_set
+
+
+class HP663X2A(SCPIMixin, Instrument):
+    """ Represents the HP / Agilent / Keysight 66312A/66332A Dynamic Measurement DC Source and
+    provides a high-level interface for interacting with the instrument.
+
+    .. code-block:: python
+
+        source = HP66312A("GPIB::1")
+        source.voltage_setpoint = 5
+        source.current_limit = 0.2
+        source.output_enabled = 1
+        print(source.current) #-> Measure current draw of load
+
+
+    """
+
+    BOOL_MAPPINGS = {True: 1, False: 0}
+
+    # Below: stop_bits: 20 comes from
+    # https://pyvisa.readthedocs.io/en/latest/api/constants.html#pyvisa.constants.StopBits
+    def __init__(self, adapter, name="HP 66312A", **kwargs):
+        super().__init__(
+            adapter,
+            name,
+            asrl={'baud_rate': 9600, 'data_bits': 8, 'parity': 0, 'stop_bits': 20},
+            **kwargs
+        )
+
+    # Source commands
+    output_enabled = Instrument.control(
+        "OUTP?", "OUTP %s",
+        """Control Enable[Disable] the source output""",
```

Comment:
Suggestion:
```suggestion
        """Control whether source output is enabled.""",
```

Also: Docstrings should end with a period (python convention).